src/change_stream/mod.rs

```diff
@@ -0,0 +1,177 @@
+//! Contains the functionality for ChangeStreams.
+pub mod document;
+pub mod option;
+
+use std::{marker::PhantomData, pin::Pin, task::Poll};
+
+use bson::Document;
+use futures::{stream::Stream, task::Context};
+use serde::{de::DeserializeOwned, Deserialize};
+
+use crate::{
+    change_stream::{
+        document::{ChangeStreamEventDocument, ResumeToken},
+        option::ChangeStreamOptions,
+    },
+    error::Result,
+    options::AggregateOptions,
+    selection_criteria::ReadPreference,
+    Client,
+    Collection,
+    Cursor,
+    Database,
+};
+
+/// A `ChangeStream` streams the ongoing changes of its associated collection, database or
+/// deployment. `ChangeStream` instances should be created with method `watch` against the relevant
+/// target.
+///
+/// `ChangeStream`'s are "resumable", meaning that they can be restarted at a given place in the
+/// stream of events. This is done automatically when the `ChangeStream` encounters certain
+/// ["resumable"](https://github.com/mongodb/specifications/blob/master/source/change-streams/change-streams.rst#resumable-error) errors, such as transient network failures. It can also be done manually by passing
```

Comment:
this line probably won't pass the linter as it is really long.

Reply:
seems like it passed!